aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -13,7 +13,56 @@ uid: fundamentals/servers/kestrel
 
 By [Tom Dykstra](https://github.com/tdykstra), [Chris Ross](https://github.com/Tratcher), and [Stephen Halter](https://twitter.com/halter73)
 
-::: moniker range=">= aspnetcore-3.0"
+::: moniker range=">= aspnetcore-5.0"
+
+Kestrel is a cross-platform [web server for ASP.NET Core](xref:fundamentals/servers/index). Kestrel is the web server that's included and enabled by default in ASP.NET Core project templates.
+
+Kestrel supports the following scenarios:
+
+* HTTPS
+* [HTTP/2](xref:fundementals/servers/kestrel/http2) (except on macOS&dagger;)
+* Opaque upgrade used to enable [WebSockets](xref:fundementals/websockets)
+* Unix sockets for high performance behind Nginx
+
+&dagger;HTTP/2 will be supported on macOS in a future release.
+
+Kestrel is supported on all platforms and versions that .NET Core supports.
+
+[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/servers/kestrel/samples/5.x) ([how to download](xref:index#how-to-download-a-sample))
+
+## Get started
+
+ASP.NET Core project templates use Kestrel by default. In *Program.cs*, the 
+<xref:Microsoft.Extensions.Hosting.GenericHostBuilderExtensions.ConfigureWebHostDefaults*> method calls <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderKestrelExtensions.UseKestrel*>:
+
+[!code-csharp[](kestrel/samples/5.x/KestrelSample/Program.cs?name=snippet_DefaultBuilder&highlight=8)]
+
+For more information on building the host, see the *Set up a host* and *Default builder settings* sections of <xref:fundamentals/host/generic-host#set-up-a-host>.
+
+## Additional resournces
+
+<a name="kestrel-options"></a>
+* <xref:fundementals/servers/kestrel/options>
+<a name="endpoint-configuration"></a>
+* <xref:fundementals/servers/kestrel/endpoints>
+<a name="http2-support"></a>
+* <xref:fundementals/servers/kestrel/http2>
+<a name="when-to-use-kestrel-with-a-reverse-proxy"></a>
+* <xref:fundementals/servers/kestrel/when-to-use-a-reverse-proxy>
+<a name="host-filtering"></a>
+* <xref:fundementals/servers/kestrel/host-filtering>
+* <xref:test/troubleshoot>
```

Comment:
In the review link, the existing references like the `<xref:test/troubleshoot>` show up as proper links but the new references `<xref:fundementals/servers/kestrel/host-filtering>` show up as plain text. Does anyone know how I can fix it?

Reply:
It's just a spelling error I think ... "fund **e** mentals"

Reply:
Others just have nit things with them ... for example ... looks like you dropped the word "support" at some point from `fundamentals/servers/kestrel/http2support`, so it should be `fundamentals/servers/kestrel/http2` now.

Best bet for the review is for me to wait until you have the build report clean. I was going to look at this first thing in the morning .... when the rotten :cat2: wakes me up at 4am no doubt! 😡 ... but it's best if I wait until the build report issues are resolved first. According to Merlin 😸, I'll be UP DAILY at 4am, so any morning is fine with him!! (**_Jerk!_** :smile:)

Reply:
btw - WRT the duplicate headings ... You have a few options there. One is use the named anchors. I don't care for them ... not HTML5-ish enough for me. I use version-specific links. For example ...

```
<h2 id="cool-title-5x">Cool title</a>

... blah blah blah ...

<h2 id="cool-title-3x">Cool title</a>

... blah blah blah ...
```

Cross-links can be versioned ...

```
::: moniker range=">= aspnetcore-5.0"

For more information, see <xref:{UID}#cool-title-5x>.

::: moniker-end

::: moniker range="< aspnetcore-5.0"

For more information, see <xref:{UID}#cool-title-3x>.

::: moniker-end

```
... or a generic cross-link without a bookmark can be used that just refers to the section by link text ...

```
For more information, see the *Cool title* section of  <xref:{UID}>.
```

They don't get the sweet direct link directly to the section, but that's very stable.